### PR TITLE
Add security to Operations

### PR DESCRIPTION
**Pull request by GraxMonzo**
## Usage

SanctumSecurityScheme:

```php
<?php

namespace App\OpenApi\SecuritySchemes;

use GoldSpecDigital\ObjectOrientedOAS\Objects\SecurityScheme;
use Vyuldashev\LaravelOpenApi\Factories\SecuritySchemeFactory;

class SanctumSecurityScheme extends SecuritySchemeFactory
{
    public function build(): SecurityScheme
    {
        return SecurityScheme::create('Sanctum')
            ->type(SecurityScheme::TYPE_HTTP)
            ->scheme('bearer');
    }
}
```
image

```php

/**
 * Gets all published car washes.
 *
 * Gets all published car washes as an array.
 */
#[OpenApi\Operation(tags: ['Car Wash'], security: ['Sanctum'])]
#[OpenApi\Parameters(factory: CommonParameters::class)]
public function index()
{
    $list = CarWash::published()
        ->orderBy('name')
        ->get();
    return response()->json($list);
}

```

image

### CODE REVIEW

**Comment by GraxMonzo** [comment]
Updated docs

**Comment by vyuldashev** [comment]
Hello. Great contribution! What if we could do both `security: ['Sanctum']` and `security: [SanctumSecurityScheme::class]`?

**Comment by GraxMonzo** [comment]
@vyuldashev Hi Vlad, thanks for your feedback. I changed it, and now you can pass both string and class as you suggested. I also created a builder for security. What do you think?

File: src/Attributes/Operation.php [hunk 1]
```diff
@@ -12,12 +14,31 @@ class Operation
...
+     * @throws InvalidArgumentException
+     */
+    public function __construct(string $id = null, array $tags = [], string $security = null, string $method = null)
     {
         $this->id = $id;
         $this->tags = $tags;
         $this->method = $method;
+
+        if ($security) {
+            $this->security = class_exists($security) ? $security : app()->getNamespace().'OpenApi\\SecuritySchemes\\'.$security.'SecurityScheme';
```
Comment (GraxMonzo) [inline src/Attributes/Operation.php:35]: I'm not sure if I did it right by hardcoding `'SecurityScheme'`.

**Comment by Kehet** [comment]
This looks awesome but could it also support OAuth2 scopes?

```# Step 1 - define the security scheme
components:
  securitySchemes:
    oAuthSample:    # <---- arbitrary name
      type: oauth2
      description: This API uses OAuth 2 with the implicit grant flow. [More info](https://api.example.com/docs/auth)
      flows:
        implicit:   # <---- OAuth flow(authorizationCode, implicit, password or clientCredentials)
          authorizationUrl: https://api.example.com/oauth2/authorize
          scopes:
            read_pets: read your pets
            write_pets: modify pets in your account

# Step 2 - apply security to individual operations
paths:
  /pets:
    patch:
      summary: Add a new pet
      security: 
        - oAuthSample: 
          - write_pets
          - read_pets
      ...
```

File: src/Attributes/Operation.php (hunk 1 above)
Comment (vyuldashev) [reply src/Attributes/Operation.php:35]: Please, change `'.$security.'SecurityScheme';` to `'.$security;`

**Comment by vyuldashev** [comment]
Released under v1.2.0

**Comment by vyuldashev** [comment]
Thanks for the contribution!

**Pull request merged**